### PR TITLE
[sp] improve column stats

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- create new component `StatsTable` to render "Quality metrics", "Statistics", and "Warnings"
- rework all column stats to use `RowDataTable` instead of `SimpleDataTable` (done in `StatsTable`)
- add new backend rates to feature set utility function
- update `RowDataTable` to alternate row backgrounds with `alternating` prop
- update `RowDataTable` to automatically add `last` prop to last `RowCard` child

# Tests
image
image

cc: @johnson-mage @nathaniel-mage @dy46

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Could you attach a screenshot in the PR if there're visual changes?

File: mage_ai/frontend/components/datasets/StatsTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import RowDataTable from '@oracle/components/RowDataTable';
+import Text from '@oracle/elements/Text';
+import { formatPercent } from '@utils/string';
+
+export type StatRow = {
+  name: string,
+  value?: any,
+  rate?: number,
+  progress?: boolean,
+  warning?: Warning;
```
Comment (dy46) [inline mage_ai/frontend/components/datasets/StatsTable/index.tsx:13]: nit: alphabetize

File: mage_ai/frontend/components/datasets/StatsTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        };
+
+        const stylePercent = (value, rate) => (
+          value !== undefined
+            ? `(${formatPercent(rate)})`
+            : formatPercent(rate)
+        );
+
+        return (
+          <RowCard columnFlexNumbers={[2,1,2]} key={name}>
```
Comment (dy46) [inline mage_ai/frontend/components/datasets/StatsTable/index.tsx:44]: spacing in array

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx [hunk 3]
```diff
@@ -4,14 +4,15 @@ import FlexContainer from '@oracle/components/FlexContainer';
 import Panel from '@oracle/components/Panel';
 import RowCard from '@oracle/components/RowCard';
 import RowDataTable from '@oracle/components/RowDataTable';
-import SimpleDataTable from '@oracle/components/Table/SimpleDataTable';
 import Spacing from '@oracle/elements/Spacing';
 import Text from '@oracle/elements/Text';
 import { COLUMN_TYPE_HUMAN_READABLE_MAPPING, COLUMN_TYPE_NUMBERS } from '@interfaces/FeatureType';
 import { PADDING_UNITS } from '@oracle/styles/units/spacing';
+import StatsTable, { StatRow } from '@components/datasets/StatsTable';
```
Comment (dy46) [inline mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:11]: fix import ordering

File: mage_ai/frontend/components/datasets/constants.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export const LARGE_WINDOW_WIDTH = 992;
+import { greaterThan, lessThan } from '@utils/array';
```
Comment (dy46) [inline mage_ai/frontend/components/datasets/constants.ts:2]: fix this

**Review by dy46 — COMMENTED** [review]
fix the styling issues

File: mage_ai/frontend/components/datasets/constants.ts [hunk 5]
```diff
@@ -50,5 +51,32 @@ export const METRICS_SORTED_MAPPING = {
...
+  ],
+  statistics: [
+    {
+      compare: greaterThan,
+      name: 'Empty columns',
+      val: 0,
+    },
+  ],
+};
+
```
[the remark this answers is not in the document]
Comment (shrey-mage) [reply mage_ai/frontend/components/datasets/constants.ts:80]: I figured it'd be easier to export all warnings under one object that could then be referenced: eg `warnings={WARNINGS.qualityMetrics}`. this object will hopefully be removed soon when I refactor the reports page to use a `StatRow[]` instead

**Review by dy46 — APPROVED** [review]

**Pull request merged**